monad-par-0.3.4.8 test suite does not compile with ghc-8.6.1

I get several errors of the following kind:

~~~~
tests/ParTests_shared.hs:88:7: error:
    • No instance for (Control.Monad.Fail.MonadFail Par)
        arising from a do statement
        with the failable pattern ‘[a, b, c, d]’
    • In a stmt of a 'do' block:
        [a, b, c, d] <- sequence [new, new, new, new]
      In the second argument of ‘($)’, namely
        ‘do [a, b, c, d] <- sequence [new, new, ....]
            fork
              $ do x <- get a
                   put b (x + 1)
            fork
              $ do x <- get a
                   put c (x + 2)
            fork
              $ do x <- get b
                   y <- get c
                   ....
            ....’
      In the expression:
        runPar
          $ do [a, b, c, d] <- sequence [new, new, ....]
               fork
                 $ do x <- get a
                      put b (x + 1)
               fork
                 $ do x <- get a
                      put c (x + 2)
               fork
                 $ do x <- get b
                      y <- get c
                      ....
               ....
   |
88 |       [a,b,c,d] <- sequence [new,new,new,new]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
~~~

--- a/monad-par/tests/ParTests_shared.hs
+++ b/monad-par/tests/ParTests_shared.hs
@@ -109,12 +109,15 @@ case_test_diamond :: Assertion
 case_test_diamond = 9 @=? (m :: Int)
  where 
   m = runPar $ do
-      [a,b,c,d] <- sequence [new,new,new,new]
-      fork $ do x <- get a; put b (x+1)
-      fork $ do x <- get a; put c (x+2)
-      fork $ do x <- get b; y <- get c; put d (x+y)
-      fork $ do put a 3
-      get d
+      abcd <- sequence [new,new,new,new]
+      case abcd of
+          [a,b,c,d] -> do
+              fork $ do x <- get a; put b (x+1)
+              fork $ do x <- get a; put c (x+2)
+              fork $ do x <- get b; y <- get c; put d (x+y)
+              fork $ do put a 3
+              get d
+          _ -> error "Oops"
 
 -- | Violate IVar single-assignment:
 --

